packages/shared/components/inputs/TokenAmountInput.svelte

```diff
@@ -42,22 +88,23 @@
 <div class="flex flex-col items-center w-full" on:click={() => amountInputElement.focus()}>
     <InputContainer {error} clearBackground clearPadding clearBorder classes="w-full flex flex-col items-center">
         <div class="flex flex-row items-end space-x-0.5">
-            <div class="flex flex-row w-full">
-                <amount-wrapper style:--max-width={`${Math.max((inputtedAmount ?? '0')?.length, 1) * 40}px`}>
+            <div class="flex flex-row w-full items-center">
+                <amount-wrapper style:--max-width={`${(inputLength * fontSize * 2) / 3}px`}>
                     <AmountInput
                         bind:inputElement={amountInputElement}
                         bind:amount={inputtedAmount}
                         maxDecimals={allowedDecimals}
+                        maxlength={maxLength}
                         isInteger={allowedDecimals === 0}
-                        fontSize={64}
+                        {fontSize}
                         clearBackground
                         clearPadding
                         clearBorder
                     />
                 </amount-wrapper>
             </div>
 
-            <Text fontWeight={FontWeight.semibold} classes="py-4">
+            <Text fontWeight={FontWeight.semibold} classes={inputLength < 14 ? 'py-4' : 'py-2'}>
```

Comment:
What's the reason fir this conditional?

Reply:
![image](https://github.com/iotaledger/firefly/assets/105642810/1aaa1ecb-f4fd-4794-9c96-9519191677c3)
here you see that the input has padding on the top and bottom. This `py-X` is here to align the `SMR` with that padding